### ISSUE TITLE
subjects are blank sometimes

### ISSUE THREAD

**Issue by drh-stanford** [sul-dlss/SearchWorks#1791]
See https://github.com/ebsco/edsapi-ruby/issues/80

Sometimes the subjects are present, but shown as blanks:

![screen shot 2017-09-15 at 9 54 31 am](https://user-images.githubusercontent.com/1861171/30494217-16cedc96-99fc-11e7-8deb-201baabc3513.png)

**Comment by bmckinney** [comment]
There is no guarantee that seachLinks will be present. This can vary from database publisher to database publisher. 

Is there something in the presenter logic that prevents subjects from being displayed without searchLink tags?

**Comment by drh-stanford** [comment]
Yes, it assumes that it's HTML that it can parse. See https://github.com/sul-dlss/SearchWorks/blob/master/app/models/concerns/eds_subjects.rb#L71-L76

Can we wrap plain subjects into `<searchLink>`'s, or return them as an array rather than faux-HTML with `<br>`'s?  The code already handles the latter case -- see https://github.com/sul-dlss/SearchWorks/blob/master/app/models/concerns/eds_subjects.rb#L68

That is,

```
eds_subjects: '<searchLink ...>subject</searchLink><br/><searchLink...>subject</searchLink>'
```

or for non-searchLink subjects

```
eds_subjects: ['subject', 'subject']
```

**Comment by bmckinney** [comment]
Ah, so it's not so much HTML parsing in general, rather the purpose is to use <searchLink> to create an array of subjects.

**Comment by drh-stanford** [comment]
Yes, I'm assuming that any subject has the metadata that a `<searchLink>` provides. So, I prefer if the gem inject a searchLink if one isn't present (i.e., it's a simple subject term).

**Comment by jkeck** [comment]
Just following up on this, it looks like the record in questions is rendering correctly now

<img width="790" alt="aph__77874260" src="https://user-images.githubusercontent.com/96776/30755523-ea7eb334-9f7b-11e7-9552-dbee752d0469.png">

Do we have another example of this issue coming up or should we close this out?

**Comment by drh-stanford** [comment]
Hmmm. It's still busted for me on the show page.

![screen shot 2017-09-25 at 10 28 07 am](https://user-images.githubusercontent.com/1861171/30822005-3f9aae2e-a1dc-11e7-8af3-afdeb658d03e.png)

**Comment by efrierson** [comment]
@cbeer @camillevilla per our conversation - i've published the [ebsco-eds 1.1.3 gem](https://rubygems.org/gems/ebsco-eds/versions/1.1.3) that will hopefully resolve the HAPI missing subject headings.  It did end up being in the gem, not in the data.  The other two are data issues further upsteam, so no other gem updates are expected to resolve those.
